### DISCUSSION TITLE
How to allow a client in guest network to connect to shairport-sync in lan network

### DISCUSSION THREAD

**Discussion by giovi321** [mikebrady/shairport-sync#1770] [Q&A]
Hi everyone,
I am using openWRT on my home router.
I have two networks:

1. lan
2. guest

The lan and guest networks cannot talk to each other (see this manual as a reference of what I have deployed: https://openwrt.org/docs/guide-user/network/wifi/guestwifi/guestwifi_dumbap )

I have added two rules in my openwrt configuration that allow any ip in guest to connect to the shairport-sync server (192.168.1.102) and allow shairport-sync server (192.168.1.102) to connect to any ip in guest. I have allowed any protocol and any port.

However, when I try to connect to shairport-sync (in lan) from my iPhone (in guest), I cannot see it in the devices list.

Any idea what am I doing wrong? Maybe the fact that they are on two separate subnets (lan = 192.168.1.0/24; guest = 192.168.100.0/24) is the problem?

**Comment by mikebrady** [comment]
Thanks for the post. AFAIK, the protocol for advertising the AirPlay service that Shairport Sync uses (it's called "Bonjour" or "ZeroConf")  is confined to the subnet, so that might account for the lack of visibility. You can use `avahi-browse` on a Linux machine to see what's being advertised. There is an app called "Discovery" from TildeSoft that allows you to do something similar on iOS and macOS.

**Comment by dubo-dubon-duponey** [comment]
mDNS/broacast traffic can cross VLANs.

How to allow that is dependent on your gear though.
Unifi has a specific configuration for that.

I am not using openWRT, but there are some blog posts around that may help you achieve this:
https://blog.christophersmart.com/2020/03/30/resolving-mdns-across-vlans-with-avahi-on-openwrt/
